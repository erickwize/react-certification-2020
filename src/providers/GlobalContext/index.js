export {
  initalState,
  GlobalContext,
  useVideList,
  useVideoInfo,
  reducer,
} from './GlobalContext.provider';
