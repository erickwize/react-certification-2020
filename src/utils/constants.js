const AUTH_STORAGE_KEY = 'wa_cert_authenticated';
const API_URL = 'https://youtube.googleapis.com/youtube/v3/';
const API_KEY = 'AIzaSyA2RrmXuKEobD96oL8gnqZz1Mh52iBUNd8';
const FAVORITE_VIDEOS_KEY = 'wa_saved_vid_list';
export { AUTH_STORAGE_KEY, API_URL, API_KEY, FAVORITE_VIDEOS_KEY };
