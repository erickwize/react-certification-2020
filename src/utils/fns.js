function random(limit) {
  return Math.floor(Math.random() * limit);
}

function getReadableDate(timeStamp) {
  const date = new Date(timeStamp);
  const options = {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  };

  return date.toLocaleDateString(undefined, options);
}

function getUsefullData(videoData) {
  const { videoId } = videoData.id;
  const {
    publishedAt,
    title,
    channelTitle,
    description,
    thumbnails,
    channelId,
  } = videoData.snippet;
  return {
    uploadDate: getReadableDate(publishedAt),
    thumbnail: thumbnails.medium.url,
    title,
    description,
    channel: channelTitle,
    channelId,
    videoId,
    link: `https://www.youtube.com/embed/${videoId}?autoplay=1&color=white`,
  };
}

export { random, getReadableDate, getUsefullData };
