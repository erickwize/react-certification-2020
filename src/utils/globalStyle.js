const color = {
  oceanBlue: '#0e4b79',
  coralPink: '#e89b93',
  lightCoral: '#fea79c',
  seaTan: '#f9c7ae',
  white: '#fff',
  black: '#000',
};

const fontSize = {
  default: '16px',
  sm: '12px',
  md: '14px',
  l: '18px',
  xxl: '20px',
  xxxl: '30px',
};

export { color, fontSize };
