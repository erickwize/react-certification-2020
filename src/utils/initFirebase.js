/*global firebase */
/* eslint-disable no-undef */

const initFirebase = () => {
  const firebaseApiKey = process.env.REACT_APP_FIREBASE_API_KEY;
  const firebaseProjectId = process.env.REACT_APP_FIREBASE_PROJECT_ID;
  const firebaseAppId = process.env.REACT_APP_FIREBASE_APP_ID;

  const firebaseConfig = {
    apiKey: firebaseApiKey,
    authDomain: `${firebaseProjectId}.firebaseapp.com`,
    databaseURL: `https://${firebaseProjectId}-default-rtdb.firebaseio.com`,
    projectId: `${firebaseProjectId}`,
    storageBucket: `${firebaseProjectId}.appspot.com`,
    messagingSenderId: "941718889832",
    appId: firebaseAppId,
    measurementId: "G-Q29WYTQKD1",
  };
  firebase.initializeApp(firebaseConfig);
};

export default initFirebase;

// const firebaseConfig = {
//   apiKey: firebaseKey,
//   authDomain: `youtubit-1adf4.firebaseapp.com`,
//   databaseURL: `https://youtubit-1adf4-default-rtdb.firebaseio.com`,
//   projectId: `youtubit-1adf4`,
//   storageBucket: `youtubit-1adf4.appspot.com`,
//   messagingSenderId: `941718889832`,
//   appId: firebaseAppId,
//   measurementId: `G-Q29WYTQKD1`,
// };

// var firebaseConfig = {
//   apiKey: `API_KEY`,
//   authDomain: `PROJECT_ID.firebaseapp.com`,
//   databaseURL: `https://PROJECT_ID.firebaseio.com`,
//   projectId: `PROJECT_ID`,
//   storageBucket: `PROJECT_ID.appspot.com`,
//   messagingSenderId: `SENDER_ID`,
//   appId: `APP_ID`,
//   measurementId: `G-MEASUREMENT_ID`,
// };
