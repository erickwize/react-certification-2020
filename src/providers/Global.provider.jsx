import React, { useContext, useReducer } from 'react';
import { reducer } from '../reducer';
import { storage } from '../utils/storage';
import { dark } from '../Themes.styles';

const theme = JSON.parse(storage.get('theme')) || dark;
const user = JSON.parse(storage.get('user')) || {authenticated:false};
const favorites = storage.get('favorites') || [];

const GlobalContext = React.createContext(null);

function useGlobal() {
  const context = useContext(GlobalContext);
  if (!context) {
    throw new Error(`Can't use "useGlobal" without an GlobalProvider!`);
  }
  return context;
}

function GlobalProvider({children}) {

  const [state, dispatch] = useReducer(reducer, {theme, search:'doggos', user, favorites, isMenuOpen:false});

  return (
    <GlobalContext.Provider value={{state, dispatch}}>
      {children}
    </GlobalContext.Provider>
  );
}

export { useGlobal };
export default GlobalProvider;