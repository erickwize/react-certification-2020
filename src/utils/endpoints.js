import { API_KEY, API_URL } from './constants';

async function fetchSearchVideos(serach = 'wizeline', videoId) {
  try {
    let queryParams = '?part=snippet&maxResults=21';
    queryParams = videoId
      ? `${queryParams}&relatedToVideoId=${videoId}&type=video`
      : `${queryParams}&q=${serach}`;

    const response = await fetch(`${API_URL}search${queryParams}&key=${API_KEY}`, {
      method: 'get',
      headers: { 'Content-Type': 'application/json' },
    });
    const data = await response.json();
    return data;
  } catch (err) {
    console.error(err);
  }
}

export { fetchSearchVideos };
