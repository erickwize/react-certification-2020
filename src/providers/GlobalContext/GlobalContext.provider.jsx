import { useState, useEffect, createContext, useReducer } from 'react';
import { color } from '../../utils/globalStyle';
import youtubeVideoList from '../../utils/mock/youtube-videos-mock.json';
import { fetchSearchVideos } from '../../api';
import { storage } from '../../utils/storage';
import { FAVORITE_VIDEOS_KEY, AUTH_STORAGE_KEY } from '../../utils/constants';

const initalStyle = {
  header: {
    background: color.oceanBlue,
    input: color.lightCoral,
  },
  card: {
    background: color.lightCoral,
    textColor: color.white,
  },
  background: `${color.white}`,
};

export const initalState = {
  user: {},
  search: 'wizeline',
  history: '',
  favoritesList: storage.get(FAVORITE_VIDEOS_KEY) || {},
  ...initalStyle,
};

export const GlobalContext = createContext(initalState);

export function useVideoInfo() {
  const [video, setVideo] = useState({});

  function updateVideoInfo(data) {
    setVideo(data);
  }

  return { video, updateVideoInfo };
}

export function reducer(state, action) {
  switch (action.type) {
    case 'SET_HISTORY':
      return {
        ...state,
        history: action.payload,
      };
    case 'SET_LIGHT_MODE':
      return {
        ...state,
        ...initalStyle,
      };
    case 'SET_DARK_MODE':
      return {
        ...state,
        header: {
          background: color.lightCoral,
          input: color.oceanBlue,
        },
        card: {
          background: color.seaTan,
          textColor: color.oceanBlue,
        },
        background: color.oceanBlue,
      };
    case 'SET_SEARCH_KEYWORD':
      return {
        ...state,
        history: state.search,
        search: action.payload,
      };
    case 'SET_FAVS_LIST':
      storage.set(FAVORITE_VIDEOS_KEY, action.payload);
      return {
        ...state,
        favoritesList: action.payload,
      };
    case 'SET_USER_INFO':
      storage.set(AUTH_STORAGE_KEY, true);
      return {
        ...state,
        user: action.payload,
      };
    case 'REMOVE_USER_INFO':
      storage.set(AUTH_STORAGE_KEY, false);
      return {
        ...state,
        user: {},
      };
    default:
      throw new Error('ACTION NOT RECOGNIZED');
  }
}

export function useVideList() {
  const [state, dispatch] = useReducer(reducer, initalState);
  const { search, history } = state;
  const [videoList, setVideoList] = useState([]);

  function updateVideoList(videos) {
    setVideoList(videos);
  }

  useEffect(() => {
    const getList = async () => {
      let listItems = [];
      dispatch({ type: 'SET_HISTORY', payload: search });
      try {
        const initalSearch = await fetchSearchVideos(search);
        listItems = initalSearch.items;
      } catch (e) {
        listItems = await youtubeVideoList.items;
        console.info('Error getting inital video search... setting youtubeVideoList');
      }
      updateVideoList(listItems);
    };
    if (search !== history) {
      getList();
    }
  }, [search, history]);

  return { videoList, updateVideoList };
}
