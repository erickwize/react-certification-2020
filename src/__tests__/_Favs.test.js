import React from 'react';
import { cleanup, render} from '@testing-library/react';
import { useHistory } from 'react-router-dom';
import UserFavorites from '../pages/Favs';
import { GlobalContext, initalState } from '../providers/GlobalContext';

import { storage } from '../utils/storage';
import { FAVORITE_VIDEOS_KEY, AUTH_STORAGE_KEY } from '../utils/constants';

const dispatch = jest.fn();
const selectCard = jest.fn();
jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

const favsList = {
  HYyRZiwBWc8: {
    channel: 'Wizeline',
    channelId: 'UCPGzT4wecuWM0BH9mPiulXg',
    description:
      'Wizeline continues to offer a Silicon Valley culture in burgeoning innovation hubs like Mexico and Vietnam. In 2018, our Guadalajara team moved into a ...',
    link: 'https://www.youtube.com/embed/HYyRZiwBWc8?autoplay=1&color=white',
    thumbnail: 'https://i.ytimg.com/vi/HYyRZiwBWc8/mqdefault.jpg',
    title: 'Wizeline Guadalajara | Bringing Silicon Valley to Mexico',
    uploadDate: 'April 18, 2019',
    videoId: 'HYyRZiwBWc8',
  },
  Po3VwR_NNGk: {
    channel: 'El Economista TV',
    channelId: 'UCXmAOGwFYxIq5qrScJeeV4g',
    description:
      'En el 2014, Bismarck fundó Wizeline, compañía tecnológica que trabaja con los corporativos ofreciendo una plataforma para que desarrollen software de forma ...',
    link: 'https://www.youtube.com/embed/Po3VwR_NNGk?autoplay=1&color=white',
    thumbnail: 'https://i.ytimg.com/vi/Po3VwR_NNGk/mqdefault.jpg',
    title: 'Wizeline hace sentir a empleados como en casa',
    uploadDate: 'March 4, 2019',
    videoId: 'Po3VwR_NNGk',
  }
}
storage.set(FAVORITE_VIDEOS_KEY, favsList);
storage.set(AUTH_STORAGE_KEY, true);

afterEach(() => {
  cleanup();
});

it('Private: Not auth', () => {
  const favs = render(
    <GlobalContext.Provider value={{ ...initalState, favoritesList: favsList }}>
      <UserFavorites selectCard={selectCard} dispatch={dispatch} />
    </GlobalContext.Provider>
  );
  expect(favs).toMatchSnapshot();
});
