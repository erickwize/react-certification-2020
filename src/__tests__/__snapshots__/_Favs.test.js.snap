// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private: Not auth 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        data-testid="favorites-view"
      >
        <h1
          class="sc-gsTCUz fNcPed"
        >
          Your favorites
        </h1>
        <p>
          2
           favorite videos
        </p>
        <section
          class="sc-bdfBwQ fGsqqP"
        >
          <div
            class="sc-dlfnbm bYLQpZ"
            data-testid="yt-videocard"
          >
            <div
              alt="Wizeline Guadalajara | Bringing Silicon Valley to Mexico"
              class="sc-gKsewC bmxZPy"
            >
              <div
                class="sc-hKgILt kQKVQt"
              >
                <div
                  class="sc-iBPRYJ bVbKde"
                  data-testid="yt-videocard-description"
                >
                  <button
                    class="sc-fubCfw kgnpdw"
                    data-testid="wa-favorite-button"
                  />
                  
                  <p
                    class="sc-jSgupP irxokA"
                  >
                    Wizeline
                     | 
                    April 18, 2019
                  </p>
                </div>
                <p
                  class="sc-eCssSg vfSIY"
                >
                  Wizeline Guadalajara | Bringing Si...
                </p>
              </div>
            </div>
          </div>
          <div
            class="sc-dlfnbm bYLQpZ"
            data-testid="yt-videocard"
          >
            <div
              alt="Wizeline hace sentir a empleados como en casa"
              class="sc-gKsewC dCeHiq"
            >
              <div
                class="sc-hKgILt kQKVQt"
              >
                <div
                  class="sc-iBPRYJ bVbKde"
                  data-testid="yt-videocard-description"
                >
                  <button
                    class="sc-fubCfw kgnpdw"
                    data-testid="wa-favorite-button"
                  />
                  
                  <p
                    class="sc-jSgupP irxokA"
                  >
                    El Economista TV
                     | 
                    March 4, 2019
                  </p>
                </div>
                <p
                  class="sc-eCssSg vfSIY"
                >
                  Wizeline hace sentir a empleados c...
                </p>
              </div>
            </div>
          </div>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      data-testid="favorites-view"
    >
      <h1
        class="sc-gsTCUz fNcPed"
      >
        Your favorites
      </h1>
      <p>
        2
         favorite videos
      </p>
      <section
        class="sc-bdfBwQ fGsqqP"
      >
        <div
          class="sc-dlfnbm bYLQpZ"
          data-testid="yt-videocard"
        >
          <div
            alt="Wizeline Guadalajara | Bringing Silicon Valley to Mexico"
            class="sc-gKsewC bmxZPy"
          >
            <div
              class="sc-hKgILt kQKVQt"
            >
              <div
                class="sc-iBPRYJ bVbKde"
                data-testid="yt-videocard-description"
              >
                <button
                  class="sc-fubCfw kgnpdw"
                  data-testid="wa-favorite-button"
                />
                
                <p
                  class="sc-jSgupP irxokA"
                >
                  Wizeline
                   | 
                  April 18, 2019
                </p>
              </div>
              <p
                class="sc-eCssSg vfSIY"
              >
                Wizeline Guadalajara | Bringing Si...
              </p>
            </div>
          </div>
        </div>
        <div
          class="sc-dlfnbm bYLQpZ"
          data-testid="yt-videocard"
        >
          <div
            alt="Wizeline hace sentir a empleados como en casa"
            class="sc-gKsewC dCeHiq"
          >
            <div
              class="sc-hKgILt kQKVQt"
            >
              <div
                class="sc-iBPRYJ bVbKde"
                data-testid="yt-videocard-description"
              >
                <button
                  class="sc-fubCfw kgnpdw"
                  data-testid="wa-favorite-button"
                />
                
                <p
                  class="sc-jSgupP irxokA"
                >
                  El Economista TV
                   | 
                  March 4, 2019
                </p>
              </div>
              <p
                class="sc-eCssSg vfSIY"
              >
                Wizeline hace sentir a empleados c...
              </p>
            </div>
          </div>
        </div>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
